### PR TITLE
Improvements/322

### PR DESCRIPTION
**Pull request by RomanGalochkin**
#322 

Please, consider previous are not merged PRs.

### CODE REVIEW

File: packages/client/src/Redirect.re [hunk 1]
```diff
@@ -0,0 +1,37 @@
+let component = ReasonReact.statelessComponent("Redirect");
+
+// @todo: fix this duplicated ligic(#rederectings)
```
Comment (OAGr) [inline packages/client/src/Redirect.re]: Do you have an idea when you are going to do this? Can you make a story for it?

File: packages/client/src/graphql/Authentication.re [hunk 2]
```diff
@@ -44,20 +40,22 @@ let component =
...
+  <QueryComponent variables=query##variables>
+    ...{({result}) =>
+      result
+      |> HttpResponse.fromApollo
+      |> HttpResponse.fmap(e => e##authentication##jwt)
+      |> (
+        e =>
+          switch (e) {
+          | Success(c) =>
+            // @todo: rapir this hard logic
```
Comment (OAGr) [inline packages/client/src/graphql/Authentication.re]: This too. Maybe some refactoring story or similar?

File: packages/client/src/App.re [hunk 3]
```diff
@@ -1,50 +1,46 @@
 type state = {
   route: Routing.Route.t,
   authToken: option(string),
+  routingCount: int,
```
Comment (OAGr) [inline packages/client/src/App.re]: Why is this needed? Can you add a comment to the code or similar? I imagine it could also have a more specific name, I really don't understand what this means.

File: packages/client/src/Redirect.re (hunk 1 above)
Comment (RomanGalochkin) [reply packages/client/src/Redirect.re]: Here I need to rethink the application structure more before doing it. For now I do not properly imagine how to do it. 

But I have added new issue with steps: 
https://github.com/OAGr/foretold/issues/434

File: packages/client/src/graphql/Authentication.re (hunk 2 above)
Comment (RomanGalochkin) [reply packages/client/src/graphql/Authentication.re]: Yes, have added new issue: https://github.com/OAGr/foretold/issues/434

File: packages/client/src/App.re (hunk 3 above)
Comment (RomanGalochkin) [reply packages/client/src/App.re]: Thanks for your question, have changed this a bit, moved this var into "Redirect" component and renamed it into "countOfRedicrectings". This var need to prevent "user cannot go from profile page".

Comment (OAGr) [reply packages/client/src/App.re]: Maybe change the name to "redirectionCount" or at least "countOfRedirections", to fix spelling?

Comment (RomanGalochkin) [reply packages/client/src/App.re]: Thanks, fixed.
